src/platform/Linux/NetworkCommissioningDriver.cpp

```diff
@@ -0,0 +1,269 @@
+    mStagingNetwork.ssidLen = static_cast<WiFiSSIDLenType>(ssid.size());
+
+    return Status::kSuccess;
+}
+
+Status LinuxWiFiDriver::RemoveNetwork(ByteSpan networkId)
+{
+    VerifyOrReturnError(networkId.size() == mStagingNetwork.ssidLen &&
+                            memcmp(networkId.data(), mStagingNetwork.ssid, mStagingNetwork.ssidLen) == 0,
+                        Status::kNetworkNotFound);
+
+    mStagingNetwork.ssidLen = 0;
+    return Status::kSuccess;
+}
+
+Status LinuxWiFiDriver::ReorderNetwork(ByteSpan networkId, uint8_t index)
+{
+    // We only support one network, so reorder is actually no-op.
+    VerifyOrReturnError(networkId.size() == mStagingNetwork.ssidLen &&
+                            memcmp(networkId.data(), mStagingNetwork.ssid, mStagingNetwork.ssidLen) == 0,
+                        Status::kNetworkNotFound);
+
+    mStagingNetwork.ssidLen = 0;
+    return Status::kSuccess;
+}
+
+void LinuxWiFiDriver::ConnectNetwork(ByteSpan networkId, ConnectCallback * callback)
+{
+    CHIP_ERROR err          = CHIP_NO_ERROR;
+    Status networkingStatus = Status::kSuccess;
+
+    // We only support one network, so reorder is actually no-op.
+    VerifyOrExit(networkId.size() == mStagingNetwork.ssidLen &&
+                     memcmp(networkId.data(), mStagingNetwork.ssid, mStagingNetwork.ssidLen) == 0,
+                 networkingStatus = Status::kNetworkNotFound);
+
+    ChipLogProgress(NetworkProvisioning, "LinuxNetworkCommissioningDelegate: SSID: %s", networkId.data());
+
+    err = ConnectivityMgrImpl().ProvisionWiFiNetwork(reinterpret_cast<const char *>(mStagingNetwork.ssid),
+                                                     reinterpret_cast<const char *>(mStagingNetwork.credentials));
+
+exit:
+
+    if (err != CHIP_NO_ERROR)
+    {
+        ChipLogError(NetworkProvisioning, "Failed to connect to WiFi network: %s", chip::ErrorStr(err));
+        networkingStatus = Status::kUnknownError;
+    }
+
+    callback->OnResult(networkingStatus, CharSpan(), 0);
+}
+
+void LinuxWiFiDriver::ScanNetworks(ByteSpan ssid, WiFiDriver::ScanCallback * callback)
+{
+    CHIP_ERROR err = DeviceLayer::ConnectivityMgrImpl().StartWiFiScan(ssid, callback);
+    if (err != CHIP_NO_ERROR)
+    {
+        callback->OnFinished(Status::kUnknownError, CharSpan(), nullptr);
+    }
+}
+
+size_t LinuxWiFiDriver::WiFiNetworkIterator::Count()
+{
+    return driver->mStagingNetwork.ssidLen == 0 ? 0 : 1;
+}
+
+bool LinuxWiFiDriver::WiFiNetworkIterator::Next(Network & item)
+{
+    if (exhausted || driver->mStagingNetwork.ssidLen == 0)
+    {
+        return false;
+    }
+    memcpy(item.networkID, driver->mStagingNetwork.ssid, driver->mStagingNetwork.ssidLen);
+    item.networkIDLen = driver->mStagingNetwork.ssidLen;
+    item.connected    = false;
+    exhausted         = true;
+    return true;
+}
+
+#endif
+
+CHIP_ERROR LinuxThreadDriver::CommitConfiguration()
+{
+    mRunningNetwork = mStagingNetwork;
+    return CHIP_NO_ERROR;
+}
```

Comment:
This should show the calls needed to do persistent storage. Does `ConnectivityMgrImpl().ProvisionWiFiNetwork` do storage? If so, then it's the opposite: RevertConfiguration will not work.